README.md

```diff
@@ -49,55 +50,65 @@ pip install git+https://github.com/keras-team/keras-cv.git tensorflow --upgrade
 ## Quickstart
 
 ```python
-import keras_cv
 import tensorflow as tf
 from tensorflow import keras
+from keras_cv.layers import RandomFlip, RandAugment, CutMix, Mixup
+from keras_cv.models import ResNet50V2Backbone, ImageClassifier
 import tensorflow_datasets as tfds
 
-# Create a preprocessing pipeline
+# Create a preprocessing pipeline with augmentations
+BATCH_SIZE = 16
+NUM_CLASSES = 3
 augmenter = keras.Sequential(
-    layers=[
-        keras_cv.layers.RandomFlip(),
-        keras_cv.layers.RandAugment(value_range=(0, 255)),
-        keras_cv.layers.CutMix(),
-        keras_cv.layers.MixUp()
+    [
+        RandomFlip(),
+        RandAugment(value_range=(0, 255)),
+        CutMix(),
+        MixUp()
     ]
 )
 
 def preprocess_data(images, labels, augment=False):
-    labels = tf.one_hot(labels, 3)
+    labels = tf.one_hot(labels, NUM_CLASSES)
     inputs = {"images": images, "labels": labels}
     outputs = augmenter(inputs) if augment else inputs
     return outputs['images'], outputs['labels']
 
-# Augment a `tf.data.Dataset`
 train_dataset, test_dataset = tfds.load(
     'rock_paper_scissors',
     as_supervised=True,
     split=['train', 'test'],
 )
-train_dataset = train_dataset.batch(16).map(
+train_dataset = train_dataset.batch(BATCH_SIZE).map(
     lambda x, y: preprocess_data(x, y, augment=True),
         num_parallel_calls=tf.data.AUTOTUNE).prefetch(
             tf.data.AUTOTUNE)
-test_dataset = test_dataset.batch(16).map(
+test_dataset = test_dataset.batch(BATCH_SIZE).map(
     preprocess_data, num_parallel_calls=tf.data.AUTOTUNE).prefetch(
         tf.data.AUTOTUNE)
 
-# Create a model
-densenet = keras_cv.models.DenseNet121(
-    include_rescaling=True,
-    include_top=True,
-    num_classes=3
+# Create a model using a pretrained backbone
+backbone = ResNet50V2Backbone.from_preset("resnet50_v2_imagenet")
+model = ImageClassifier(
+    backbone=backbone,
+    num_classes=3,
+    activation="softmax",
 )
-densenet.compile(
+model.compile(
     loss='categorical_crossentropy',
-    optimizer='adam',
+    optimizer=keras.optimizers.Adam(learning_rate=1e-5),
     metrics=['accuracy']
 )
 
+# Freeze backbone to avoid overfitting to small dataset
+model.backbone.trainable = False
```

Comment:
Doesn't it require to set `model.compile` right after changing the trainable attributes?

Reply:
Worked just fine in my colab (looking at `model.summary()`). Will add a gist once I respond to some comments

Reply:
From the [doc](https://keras.io/guides/transfer_learning/)

![image](https://user-images.githubusercontent.com/17668390/236407319-879ab6b4-0734-4c46-8979-60d131d5de9f.png)

The `model.summary` might have possible bugs, relevant [ticket](https://github.com/keras-team/keras/issues/18092). Mentioning the author of the above doc to give comment on it, cc. @fchollet

Reply:
@innat I guess bug happens when you try to freeze the whole model, when there are nested models they behave like a layer so `model.backbone.trainable` may not be an issue (not tested).

Reply:
@Frightera 
Yes, I think so. 

Now, if we don't recompile the model after changing its trainable attributes, what would happen in real? How it would affect the rest of the code? Does the model summary report something useful? 

[similar [feedback](https://github.com/keras-team/keras-io/pull/1314#discussion_r1164088972) cc. @martin-gorner]

Reply:
@innat Model summary should show something useful if you freeze and unfreeze layers like this:
```python
for layer in model.layers[0:10]:
    layer.trainable = True
```
But I am not sure if they are tracked properly without calling `compile()`. Needs further testing I guess.

Reply:
Thanks this is a great catch @innat! The model does not train very well with a frozen backbone actually 🥶 🥶 🥶

Reply:
> The model does not train very well with a frozen backbone actually 

I think it's a different thing from what I was concerned about. I wondered about 

```python
# using!
model = keras.Sequentail([backbone, neck, head])
model.compile( ... )
model.backbone.trainable = Fasle

# or should it be this way!
model = keras.Sequentail([backbone, neck, head])
model.backbone.trainable = Fasle
model.compile( ... )
```

Reply:
No, we are agreeing @innat! My previous runs had your first setup and the backbone was actually training. When I switched to the second the model behaved totally differently and I verified that the backbone was actually frozen. So
* only the second approach works
* a frozen backbone doesn't seem to work well (or would need to some fiddling to work well) in my quickstart